akka-stream/src/main/scala/akka/stream/IOResult.scala

```diff
@@ -48,3 +50,10 @@ object IOResult {
   def createFailed(count: Long, ex: Throwable): IOResult =
     new IOResult(count, Failure(ex))
 }
+
+/**
+ * This exception signals that a stream has been completed by an onError signal
+ * while there was still IO operations in progress.
+ */
+final case class AbruptIOTerminationException(ioResult: IOResult, cause: Throwable)
```

Comment:
Ah, neat providing the IOResult even on failure.

Reply:
Yea. It could be the case that some data has already been written to the file when stream fails. So I figured it would be good to still expose that information.